app/src/gui/components/notebook/record_table.tsx

```diff
@@ -546,35 +547,55 @@ function RecordsTable(props: RecordsTableProps) {
  * @returns {JSX.Element} The rendered table for browsing records.
  */
 export function RecordsBrowseTable(props: RecordsBrowseTableProps) {
-  const {recordLabel} = props;
+  const {recordLabel, filterByUser} = props;
   // TODO validate this is always defined
   const activeToken = useAppSelector(selectActiveUser)!.parsedToken;
-
+  const activeUser = useAppSelector(selectActiveUser);
   const [query, setQuery] = React.useState('');
+
   const {data: records, isLoading: recordsLoading} = useQuery({
-    queryKey: ['allrecords', query, props.project_id],
+    queryKey: ['allrecords', query, props.project_id, filterByUser],
     networkMode: 'always',
     gcTime: 0,
     queryFn: async () => {
       if (query.length === 0) {
-        console.log(
-          'Getting metadata for all records with project ID',
-          props.project_id
-        );
         const rows = await getMetadataForAllRecords(
           activeToken,
           props.project_id,
           props.filter_deleted
         );
```

Comment:
Just FYI this method already filters against permissions - the user will only see other's records if they have permission to - eventually we will probably want to not even show the All responses tab unless the other has permission to see other people's responses.